client/lib/wp/sync-handler/index.js

```diff
@@ -187,13 +193,48 @@ export class SyncHandler {
 		return localforage.getItem( key, fn );
 	}
 
+	/**
+	 * Add/Override a record.
+	 *
+	 * @param {String} key - record key identifier
+	 * @param {Object} data - data to store
+	 * @param {Function} [fn] - callback function
+	 * @return {Promise} natuve ES6 promise
+	 */
 	storeRecord( key, data, fn = () => {} ) {
 		debug( 'storing data in %o key\n', key );
-		return localforage.setItem( key, data, fn );
+
+		// add this record to history
+		return cacheInvalidation
```

Comment:
> ```es6
const cache = new MaxAgeCache();
```

Max age won't the only one reason to clean cache. Next step in this sense is to prune records from posts list when the pagination has shifted for instance.

> ```es6
cache.addItem( key, data );
```

I like `cache` instead of `cacheInvalidation`